src/corona_hakab_model/consts.py

```diff
@@ -3,121 +3,119 @@
+    "asymptomatic_infection_ratio": 0.25,
+    # probability of an infected silent agent infecting others
+    "silent_infection_ratio": 0.3,  # todo i made this up, need to get the real number
+    # base r0 of the disease
+    "r0": 2.4,
+    # isolation policy
+    # todo why does this exist? doesn't the policy set this? at least make this an enum
+    # note not to set both home isolation and full isolation true
+    # whether to isolation detected agents to their homes (allow familial contact)
+    "home_isolation_sicks": False,
+    # whether to isolation detected agents fully (no contact)
+    "full_isolation_sicks": False,
+    # how many of the infected agents are actually caught and isolated
+    "caught_sicks_ratio": 0.3,
+    # policy stats
+    # todo this reeeeally shouldn't be hard-coded
+    # defines whether or not to apply a isolation (work shut-down)
+    "active_isolation": False,
+    # the date to stop work at
+    "stop_work_days": 30,
+    # the date to resume work at
+    "resume_work_days": 60,
+    # social stats
+    # the average family size
+    "average_family_size": 5,  # todo replace with distribution
+    # the average workplace size
+    "average_work_size": 50,  # todo replace with distribution
+    # the average amount of stranger contacts per person
+    "average_amount_of_strangers": 200,  # todo replace with distribution
+    # relative strengths of each connection (in terms of infection chance)
+    # todo so if all these strength are relative only to each other (and nothing else), whe are none of them 1?
+    "family_strength_not_workers": 0.75,
+    "family_strength": 0.4,
+    "work_strength": 0.04,
+    "stranger_strength": 0.004,
+}
+
+ConstParameters = namedtuple('ConstParameters', sorted(default_parameters))
+
+
+class Consts(ConstParameters):
+
+    @staticmethod
+    def default():
+        return Consts(**default_parameters)
+
+    @staticmethod
+    def from_file(param_path):
         """
+        Load parameters from file and return Consts object with those values.
 
-        valid_keys = self.simulation_parameters.keys()
-
-        for key in dct.keys():
-            if key in valid_keys:
-                if key == "distributions":  # Special care here
-                    self.set_distribution_params(dct["distributions"])
-                else:
-                    setattr(self, key, dct[key])
-            else:
-                self.logger.critical(f"Unknown parameter name: {key}")
-                raise UnknownParameterException(key)
+        We sanitize the loaded data, for obvious reasons.
+        Documentation about what is allowed and not allowed can be found at the top of this page.
+        """
+        with open(param_path, "r") as read_file:
+            data = read_file.read()
 
-    def set_distribution_params(self, distributions_dict):
-        valid_distribution_keys = self.simulation_parameters["distributions"].keys()
+        Consts.sanitize_data(data)
```

Comment:
as far as I can tell this function doesn't exist

Reply:
6 lines down :)